real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/main/RobobuggyConfigFileTest.java

```diff
@@ -0,0 +1,206 @@
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import java.io.*;
+
+/**
+ * Created by davidneiman on 9/2/16.
+ */
+public class RobobuggyConfigFileTest {
+
+    /**
+     * Sets up test cases
+     *  @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
+        //Runs before each test case, denoted "@Test" before the method
+    }
+
+    /**
+     * Cleans up test cases
+     * @throws Exception
+     */
+    @After
+    public void tearDown() throws Exception {
+        //Runs after each test case
+    }
+
+    /**
+     * Creates a config file, setting the ROBOT field to the given input
+     * @param s: String to be used as the robot type
+     * @throws IOException
+     */
+    public void createConfigFileWithRobot(String s) throws IOException{
+
+        //BufferedWriter writer = new BufferedWriter(new FileWriter(new File("config.config")));
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(new File("config.config")), "UTF-8"));
+
+        writer.write("{" +
+                "\"COM_PORT_IMU\":\"NOT_SET\"," +
+                "\"COM_PORT_RBSM\":\"NOT_SET\"," +
+                "\"PORT_VISION\":\"NOT_SET\"," +
+                "\"COM_PORT_GPS\":\"NOT_SET\"," +
+                "\"WAYPOINT_SOURCE_LOG_FILE\":\"config.config\"," +
+                "\"PLAY_BACK_SOURCE_FILE\":\"NOT_SET\"," +
+                "\"VIDEO_FILE\":\"NOT_SET\"," +
+                "\"ROBOT_TYPE\":\"");
+        writer.write(s);
+        writer.write("\"}");
+        writer.close();
+    }
+
+    /**
+     * Creates a config file without a ROBOT field
+     * Use only for testing; it's supposed to lead to null-pointer exceptions
+     * @throws IOException
+     */
+    public void createConfigFileWithoutRobot() throws IOException{
+        //BufferedWriter writer = new BufferedWriter(new FileWriter(new File("config.config")));
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(new File("config.config")), "UTF-8"));
+
+        writer.write("{" +
+                "\"COM_PORT_IMU\":\"NOT_SET\"," +
+                "\"COM_PORT_RBSM\":\"NOT_SET\"," +
+                "\"PORT_VISION\":\"NOT_SET\"," +
+                "\"COM_PORT_GPS\":\"NOT_SET\"," +
+                "\"WAYPOINT_SOURCE_LOG_FILE\":\"config.config\"," +
+                "\"PLAY_BACK_SOURCE_FILE\":\"NOT_SET\"," +
+                "\"VIDEO_FILE\":\"NOT_SET" +
+                "\"}");
+        writer.close();
+    }
+
+
+    /**
+     * Tests basic get/set functionality for a normal config file
+     * Doesn't actually create any robots
+     */
+    @Test
+    public void testRobotTypeGetSet() {
+
+        //Create a new config file with a reasonable robot type
+        try {
+            createConfigFileWithRobot("TransistorAuton");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
```

Comment:
You can combine multiple `catch` blocks if they do the same thing, change this (and others) to `IOException | IllegalArgumentException e`